### PR TITLE
doc/start: remove mention of Centos 8 support

### PR DESCRIPTION
**Pull request by zdover23**
Remove information so that it is clear that Centos 8 will no longer be supported by Ceph.

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [x] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [x] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

**Comment by zdover23** [comment]
@anthonyeleven: Don't merge this one, even if it's approved. The Leadership Team is going to have a discussion about... well, about various things here.

File: doc/start/os-recommendations.rst
Comment (idryomov) [inline doc/start/os-recommendations.rst:52]: Should Centos 9 line receive an A for Quincy?

Comment (cbodley) [reply doc/start/os-recommendations.rst:52]: "C" probably? we've been building centos 9 packages for quincy but it's not part of our qa: https://github.com/ceph/ceph/tree/quincy/qa/distros/supported

**Review by adk3798 — APPROVED** [review]
Only concern was the one already mentioned in https://github.com/ceph/ceph/pull/58092#discussion_r1643050591

**Review by dmick — APPROVED** [review]

**Comment by tchaikov** [comment]
@zdover23 so we don't need to support CentOS 8 ? right? i am asking, because i had a PR got reverted because of the incompatibility with the Python shipped along with CentOS 8.

**Comment by zdover23** [comment]
> @zdover23 so we don't need to support CentOS 8 ? right? i am asking, because i had a PR got reverted because of the incompatibility with the Python shipped along with CentOS 8. and the PR addresses the incompatibility with Python 3.12, which is the default Python in fedora 40.

@tchaikov, I'm not sure what the answer to this is, but I am sure that I can't unilaterally say what it is. I think this is a question for the entire leadership team.

**Comment by tchaikov** [comment]
@zdover23 thank you for your quick reply. thought this PR was a followup of the consensus of the team. anyway, will send a mail to the mailing list.

**Comment by anthonyeleven** [comment]
Show of hands:  enterprises that are running critical production loads on Fedora 40?

**Comment by dmick** [comment]
@anthonyeleven why is Fedora 40 at issue here?

**Comment by cbodley** [comment]
> we don't need to support CentOS 8 ? right? i am asking, because i had a PR got reverted because of the incompatibility with the Python shipped along with CentOS 8

@tchaikov the intent was to drop centos 8 support for squid, but our container builds only very recently stopped depending on centos 8. so now is probably a good time to revisit your pr

File: doc/start/os-recommendations.rst
Comment (ljflores) [reply doc/start/os-recommendations.rst:52]: Agreed with @cbodley

Comment (ljflores) [reply doc/start/os-recommendations.rst:52]: FYI here is the PR adding c9 support to the quincy suite for those interested in reviewing: https://github.com/ceph/ceph/pull/58104

**Review by ljflores — COMMENTED** [review]
Agreed with @cbodley and @idryomov that quincy should get a C for c9 since we've been building c9 packages, but they're untested.

**Comment by idryomov** [comment]
> Agreed with @cbodley and @idryomov that quincy should get a C for c9 since we've been building c9 packages, but they're untested.

I think it should get an A with a note -- something like "A (since 17.2.8)".

**Comment by tchaikov** [comment]
> Show of hands: enterprises that are running critical production loads on Fedora 40?

the reasons why i mentioned fedora 40:

- as an open source project, our downstream distros are packaging the source tarball of our latest releases. and these distros are likely using the latest stable releases of Python as well. if Ceph does not run or even build with latest Python, it causes headache of maintainers, and hurts the community and the ecosystem.
- fedora is the upstream of CentOS and its derivative distros. fedora 40 will be the base of a future version of CentOS, so your are correct that "enterprises that are running critical production loads on Fedora 40". if you believe that enterprises should be running a distro with "Enterprise" in its name, which i doubt. but that's probably not in the scope of the discussion.
- some Python APIs were deprecated, and then removed from Python.  and fedora 40 uses latest Python. so to be compatible with the latest stable release of fedora is a good test for compatibility with the latest Python. so the tree is future proof. -- we don't have to worry about the fail to build from source and/or the not well-tested build shipped next week.

**Comment by tchaikov** [comment]
> > we don't need to support CentOS 8 ? right? i am asking, because i had a PR got reverted because of the incompatibility with the Python shipped along with CentOS 8
> 
> @tchaikov the intent was to drop centos 8 support for squid, but our container builds only very recently stopped depending on centos 8. so now is probably a good time to revisit your pr

Casey, as always. much appreciated. i will send it for review. i noticed that we had https://tracker.ceph.com/issues/66399 the other day. and wanted to fix it.

**Comment by anthonyeleven** [comment]
What I believe isn't material:  I note the reality of the people *who actually use and need Ceph*.

>fedora 40 will be the base of a future version of CentOS

Of Stream.  Many companies want nothing of Stream and plan to go elsewhere by or after CentOS 9.  Or stay there indefinitely.

**Comment by tchaikov** [comment]
> What I believe isn't material: I note the reality of the people _who actually use and need Ceph_.
> 
> > fedora 40 will be the base of a future version of CentOS
> 
> Of Stream. Many companies want nothing of Stream and plan to go elsewhere by or after CentOS 9. Or stay there indefinitely.

please kindly note, CentOS is not coming from nowhere.

**Comment by zdover23** [comment]
@ljflores, are we ready to merge this after I make the changes suggested here, or should I leave this open for another week?

**Review by anthonyeleven — APPROVED** [review]

**Review by ktdreyer — APPROVED** [review]

File: doc/start/os-recommendations.rst
Comment (idryomov) [reply doc/start/os-recommendations.rst:52]: Note from today's CLT meeting:
> - https://github.com/ceph/ceph/pull/58092 (Zac)
>   - as no more discussion on moving away from centos has taken place,
>    and for now we are effectively replacing centos 8 with centos 9 in
>    the suites, centos 9 should be listed as tested (status A, starting
>    from 17.2.8 for quincy and 18.2.3 for reef)

A lot of people have approved the PR as is, but IMO this should be incorporated here to avoid it slipping through the cracks in the coming weeks.

**Comment by ljflores** [comment]
@zdover23 there are enough approvals, I think if all requested changes are addressed it can be merged.

**Comment by zdover23** [comment]
> @zdover23 there are enough approvals, I think if all requested changes are addressed it can be merged.

@ljflores I'll take care of it within the hour.

**Comment by zdover23** [comment]
I've given Quincy-Centos 9 a "C". When we begin testing, I'll change it to an "A".

File: doc/start/os-recommendations.rst
Comment (zdover23) [reply doc/start/os-recommendations.rst:52]: @neha-ojha

Comment (neha-ojha) [reply doc/start/os-recommendations.rst:52]: +1

File: doc/start/os-recommendations.rst
Comment (cbodley) [inline doc/start/os-recommendations.rst:75]: looks good for quincy, but reef was already an A for centos 9. that part shouldn't need to be qualified

Comment (zdover23) [reply doc/start/os-recommendations.rst:75]: Roger that. I'll take it out.

Comment (idryomov) [reply doc/start/os-recommendations.rst:75]: > reef was already an A for centos 9. that part shouldn't need to be qualified

Right, my apologies for confusion -- I got too carried away with upgrade suites in my note (we clearly weren't testing _that_ on centos 9).

Since I'm commenting already, I wonder why centos 8 line isn't just being removed?  Keeping around an empty line doesn't make sense to me.

**Review by idryomov — APPROVED** [review]

**Comment by zdover23** [comment]
https://github.com/ceph/ceph/pull/58389 - Squid backport
https://github.com/ceph/ceph/pull/58390 - Reef backport
https://github.com/ceph/ceph/pull/58391 - Quincy backport

**Pull request merged**
